src/jep/convert_p2j.c

```diff
@@ -52,6 +56,58 @@ static jmethodID unmodifiableList      = 0;
     static jstring UTF8 = NULL;
 #endif
 
+bool isFunctionalInterfaceType(JNIEnv *env, jclass type) {
+    if (!JNI_METHOD(classGetMethods, env, JCLASS_TYPE, "getMethods", "()[Ljava/lang/reflect/Method;")) {
+        process_java_exception(env);
+        return NULL;
+    }
+    if (!JNI_METHOD(classGetModifiers, env, JCLASS_TYPE, "getModifiers", "()I")) {
+        process_java_exception(env);
+        return NULL;
+    }
+    if (!JNI_METHOD(methodGetModifiers, env, JMETHOD_TYPE, "getModifiers", "()I")) {
+        process_java_exception(env);
+        return NULL;
+    }
+    if ((*env)->PushLocalFrame(env, JLOCAL_REFS) != 0) {
+        process_java_exception(env);
+        return NULL;
+    }
+    int classModifiers = (*env)->CallIntMethod(env, type, classGetModifiers);
+    if (!(classModifiers & 512)) {
```

Comment:
Any reason to not use `Class.isInterface()` here instead of `Class.getModifiers()` and comparing the modifier value?